Storage/History.swift

```diff
@@ -40,6 +37,11 @@ public protocol BrowserHistory {
     func getPinnedTopSites() -> Deferred<Maybe<Cursor<Site>>>
 }
 
+/** An interface for fast repeated frecency queries. */
+public protocol OptimizedFrecency {
+    func getSites(historyLimit limit: Int, bookmarksLimit: Int, whereURLContains filter: String?, groupClause: String, whereData: String?) -> Deferred<Maybe<Cursor<Site>>>
```

Comment:
If I were to implement `BrowserHistory` on a non-SQL backend, that `groupClause` and `whereData` become meaningless. That means this is a dodgy abstraction — the caller of this public method needs to know the internal SQL table structures of the implementation, and write some SQL fragments to match.

Try it: go implement `MockHistory` rather than `fatalError`ing!

Reply:
Agreed, I won't keep this.